### PR TITLE
op-challenger: Fix prestate checks

### PR DESCRIPTION
**Pull request by ajsutton**
**Description**

Fix absolute pre-state validation in `op-challenger`. Previously the output root and cannon prestates from the contract were both compared to the local output root prestate.  However this check was only done for already completed games.

The devnet now updates the cannon absolute prestate hash instead of hard coding an old one in the template.

Since the devnet deploys the fault proof system in L1 genesis, the L2 genesis output root can't be known at deploy time. To allow the challenger to still be tested, a new hidden option `--allow-invalid-prestate` is added which causes it to log an error when the prestate doesn't match but continue to play the game.

**Tests**

Added unit tests.

**Metadata**

- Fixes https://github.com/ethereum-optimism/client-pod/issues/561

### CODE REVIEW

File: op-challenger/game/fault/validator.go [hunk 1]
```diff
@@ -40,7 +42,7 @@ func (v *PrestateValidator) Validate(ctx context.Context) error {
 		return fmt.Errorf("failed to fetch provider's prestate hash: %w", err)
 	}
 	if !bytes.Equal(prestateCommitment[:], prestateHash[:]) {
-		return fmt.Errorf("provider's absolute prestate does not match contract's absolute prestate: Provider: %s | Contract: %s", prestateCommitment.Hex(), prestateHash.Hex())
+		return fmt.Errorf("provider's %v absolute prestate does not match contract's absolute prestate: Provider: %s | Contract: %s", v.valueName, prestateCommitment.Hex(), prestateHash.Hex())
```
Comment (refcell) [inline op-challenger/game/fault/validator.go]: The `valueName` is passed in as `alphabet`, `cannon`, and `output root`. Should this instead be

```suggestion
		return fmt.Errorf("%v provider's absolute prestate does not match contract's absolute prestate: Provider: %s | Contract: %s", v.valueName, prestateCommitment.Hex(), prestateHash.Hex())
```

Comment (ajsutton) [reply op-challenger/game/fault/validator.go]: Both ways work depending on how you think about it.  If the trace provider is the combined one then `provider's cannon absolute prestate` is right and if the provider is the nested one then `cannon provider's absolute prestate` is right.

Comment (ajsutton) [reply op-challenger/game/fault/validator.go]: Have switched it round though.

File: op-challenger/game/scheduler/coordinator.go [hunk 2]
```diff
@@ -133,7 +135,10 @@ func (c *coordinator) createJob(ctx context.Context, game types.GameMetadata, bl
 			return nil, fmt.Errorf("failed to create game player: %w", err)
 		}
 		if err := player.ValidatePrestate(ctx); err != nil {
-			return nil, fmt.Errorf("failed to validate prestate: %w", err)
+			if !c.allowInvalidPrestate {
+				return nil, fmt.Errorf("failed to validate prestate: %w", err)
+			}
+			c.logger.Error("Invalid prestate", "game", game.Proxy, "err", err)
```
Comment (Inphi) [inline op-challenger/game/scheduler/coordinator.go:141]: nit: This skips over all validation errors, including RPC-related errors. Probably better to only ignore specifically prestate mismatches.

**Review by Inphi — APPROVED** [review]
LGTM!

File: op-challenger/game/scheduler/coordinator.go (hunk 2 above)
Comment (ajsutton) [reply op-challenger/game/scheduler/coordinator.go:141]: The RPC calls are only done to check the prestate which we'd ignore but yeah its probably worth being picky so we test as much of the code path as we can in CI.

**Pull request merged**